src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/resolver/JsonScalarResolverTest.kt

```diff
@@ -0,0 +1,68 @@
+package it.krzeminski.snakeyaml.engine.kmp.resolver
+
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.shouldBe
+import it.krzeminski.snakeyaml.engine.kmp.nodes.Tag
+
+class JsonScalarResolverTest: FunSpec({
+    val scalarResolver = JsonScalarResolver()
+
+    test("resolve explicit scalar") {
+        scalarResolver.resolve("1", implicit = false) shouldBe Tag.STR
+    }
+
+    test("resolve implicit integer") {
+        scalarResolver.resolve("1", implicit = true) shouldBe Tag.INT
```

Comment:
suggestion: we have lots of assertions inside a single test. If the first one fails we won't know the result for other assertions. If we could wrap the assertions inside `assertSoftly` it would solve that inconvenience

Reply:
https://github.com/krzema12/snakeyaml-engine-kmp/pull/399/commits/2d0180b8094b1b8c5a222e64f7d6e1532ef07810